kpi/models/asset.py

```diff
@@ -682,17 +690,15 @@ def latest_deployed_version(self):
 
     @property
     def version_id(self):
-        try:
-            latest_versions = self.prefetched_latest_versions
-        except AttributeError:
-            latest_version = self.latest_version
-            if latest_version:
-                return latest_version.uid
-        else:
-            try:
-                return latest_versions[0]
-            except IndexError:
-                return None
+        latest_version = self.latest_version
+        if latest_version:
+            return latest_version.uid
```

Comment:
I almost forgot, but yes :) Unless the versions have been prefetched, the `latest_version` property executes a database query: https://github.com/kobotoolbox/kpi/pull/1961/files/33fe0f8e7ff714bdd33b600d0cf77f411603163c#diff-43986855ae1015d5975b107cf1ee9869R676. The intermediate variable was to avoid doubling the query.

Reply:
I understand.